### PR TITLE
Logout feature

### PR DESCRIPTION
**Pull request by mariana0412**
- Added a logout icon in the top-left corner of the navigation bar of the `SetupView`
- Integrated `UserStateViewModel` into `IOSAccessAssessmentApp` as a shared EnvironmentObject for auth state management
- Added a confirmation dialog after the logout button tap for better UX
- Introduced constants enum in `SetupView` for better code maintainability
- Removed duplicate "Setup view" label at the Setup screen, left only "Setup"

### CODE REVIEW

File: IOSAccessAssessment/IOSAccessAssessmentApp.swift
Comment (himanshunaidu) [inline IOSAccessAssessment/IOSAccessAssessmentApp.swift:12]: Oh I didn't notice this while reviewing the login PR. 
This does help quite a few things. Nice one!

File: IOSAccessAssessment/ViewModel/UserStateViewModel.swift
Comment (himanshunaidu) [inline IOSAccessAssessment/ViewModel/UserStateViewModel.swift:1]: Oof, too many folders, but I guess this makes sense with the MVVM architecture (although that's not something we are using throughout the app... maybe a second round of refactoring is in order once all the main objectives RAI program are done).

File: IOSAccessAssessment/Views/SetupView.swift
Comment (himanshunaidu) [inline IOSAccessAssessment/Views/SetupView.swift:12]: Again, I do believe it would be good to centralize the constants, even if they are hard-tied to some views at this moment. 
Maybe we can change the 'Model' folder (which tbh isn't serving as a Model folder anymore) to 'Constants', and house the main constants over there.

**Comment by himanshunaidu** [comment]
Overall, this is great.
I am fine with merging this one now itself, which I guess may cause conflicts with the change set pr, but that one imo should be merged last anyway. 
But merging this is your call. Maybe you can deal with the comments now itself, or choose to do so in a separate PR.

**Review by himanshunaidu — APPROVED** [review]

File: IOSAccessAssessment/Views/SetupView.swift
Comment (himanshunaidu) [reply IOSAccessAssessment/Views/SetupView.swift:71]: The confirmation dialog doesn't show any content. Randomly clicking somewhere around the confirmation dialog finally logs me out. 

https://github.com/user-attachments/assets/9e753340-38a6-4cf4-8156-27db0ae949dd


May I know how this confirmation dialog looks on iPhone?

File: IOSAccessAssessment/IOSAccessAssessmentApp.swift
Comment (himanshunaidu) [reply IOSAccessAssessment/IOSAccessAssessmentApp.swift:12]: I am having some second thoughts on if ViewModel is something we should go ahead with, after having most of the application being implemented in (sort of) MV pattern.

Comment (himanshunaidu) [reply IOSAccessAssessment/IOSAccessAssessmentApp.swift:12]: This thread is mostly just a thought experiment I'd like to discuss whenever possible. 
I won't be blocking the PR just for this.

File: IOSAccessAssessment/IOSAccessAssessmentApp.swift
Comment (himanshunaidu) [inline IOSAccessAssessment/IOSAccessAssessmentApp.swift:12]: I wonder if this way of setting up the authentication would be the best way to go. At the first glance, I do not see a reason to store authentication status in user defaults, when we can keep checking if the auth tokens are valid.  
May I be reminded on the incentive behind shifting to AppStorage instead of ViewModel?

Here are some doubts I have
1. Without a valid refresh token I am not sure if we can save any valid authentication. 
2. Even if we can enable user auth persistence, shouldn't we be continuously checking if the KeyChain still has valid credentials, instead of caching isAuthenticated to an insecure storage?

**Review by himanshunaidu — DISMISSED** [review]

File: build/XCBuildData/fc98efb9a669f35cfc6243fbfcb55a1f.xcbuilddata/build-request.json
Comment (himanshunaidu) [inline build/XCBuildData/fc98efb9a669f35cfc6243fbfcb55a1f.xcbuilddata/build-request.json:1]: Hello, I don't want to block the PR any longer, but may I know what's the deal with this build folder?
How are these being generated?
I don't think these should be here.

File: IOSAccessAssessment.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (himanshunaidu) [inline IOSAccessAssessment.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: May I know why this file was deleted?
The documentation online makes me believe this is an important file for future debugging.

File: build/XCBuildData/fc98efb9a669f35cfc6243fbfcb55a1f.xcbuilddata/build-request.json
Comment (himanshunaidu) [reply build/XCBuildData/fc98efb9a669f35cfc6243fbfcb55a1f.xcbuilddata/build-request.json:1]: Yeah, unfortunately these files should definitely not be here.

**Comment by himanshunaidu** [comment]
Functionality-wise, this works great. 
Please just get rid of the build files, let me know if it is wise to remove the Package.resolved file, and we can merge it.

File: build/XCBuildData/fc98efb9a669f35cfc6243fbfcb55a1f.xcbuilddata/build-request.json
Comment (mariana0412) [reply build/XCBuildData/fc98efb9a669f35cfc6243fbfcb55a1f.xcbuilddata/build-request.json]: Thanks for catching that! Yes, the `build/ `folder is generated during the build process and shouldn’t be in the repo. I’ve updated the `.gitignore` to prevent committing it in future.

File: IOSAccessAssessment.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (mariana0412) [reply IOSAccessAssessment.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: I may have unintentionally deleted it while resolving my local issues with the swift-collections dependency. This file is a record of the exact versions of dependencies used in the project and it is regenerated by SPM, so it should not be a problem. In this .gitignore template https://github.com/github/gitignore/blob/main/Swift.gitignore they say "Add this line if you want to avoid checking in source code from Swift Package Manager dependencies." But if we want to ensure consistency in dependency versions across all environments, I can restore it.

**Comment by himanshunaidu** [comment]
Alright! Looks good to me.
Thanks for resolving this

**Review by himanshunaidu — APPROVED** [review]

**Pull request merged**
